commit d5d375b6386f493e2ebc3866805e6a52eb34a2ae
Author:     Christian Tacke <c.tacke@gsi.de>
AuthorDate: Fri Jun 30 12:52:27 2023 +0200
Commit:     Christian Tacke <c.tacke@gsi.de>
CommitDate: Fri Jun 30 12:52:27 2023 +0200

    debian/patches: Add fix-missing-unlock-in-sasl_do_kinit_ext_cc.patch
    
    Patch from upstream.
    
    Do not hang in kerberos authenticated ldap.

diff --git a/debian/changelog b/debian/changelog
index 5ff4f1c..baf9910 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,12 @@
+autofs (5.1.8-3) UNRELEASED; urgency=medium
+
+  [ Christian Tacke ]
+  * debian/patches:
+    * Add fix-missing-unlock-in-sasl_do_kinit_ext_cc.patch from upstream.
+      Do not hang in kerberos authenticated ldap.
+
+ -- Mike Gabriel <sunweaver@debian.org>  Fri, 30 Jun 2023 12:06:48 +0200
+
 autofs (5.1.8-2) unstable; urgency=medium
 
   [ Mike Gabriel ]
diff --git a/debian/patches/fix-missing-unlock-in-sasl_do_kinit_ext_cc.patch b/debian/patches/fix-missing-unlock-in-sasl_do_kinit_ext_cc.patch
new file mode 100644
index 0000000..7482240
--- /dev/null
+++ b/debian/patches/fix-missing-unlock-in-sasl_do_kinit_ext_cc.patch
@@ -0,0 +1,41 @@
+Origin: https://git.kernel.org/pub/scm/linux/storage/autofs/autofs.git/commit/?id=b2571ed0df973a6dc6a8e661874655fa7cecdc37
+
+commit b2571ed0df973a6dc6a8e661874655fa7cecdc37
+Author: James Dingwall <james-autofs@dingwall.me.uk>
+Date:   Wed Jul 20 13:22:38 2022 +0800
+
+    autofs-5.1.8 - fix missing unlock in sasl_do_kinit_ext_cc()
+    
+    There is a missing mutex unlock in function sasl_do_kinit_ext_cc(),
+    fix it.
+    
+    Signed-off-by: James Dingwall <james-autofs@dingwall.me.uk>
+    Signed-off-by: Ian Kent <raven@themaw.net>
+
+# diff --git a/CHANGELOG b/CHANGELOG
+# index 1f7c93a..e0b285d 100644
+# --- a/CHANGELOG
+# +++ b/CHANGELOG
+# @@ -27,6 +27,7 @@
+#  - add autofs_strerror_r() helper for musl.
+#  - update configure.
+#  - handle innetgr() not present in musl.
+# +- fix missing unlock in sasl_do_kinit_ext_cc().
+#  
+#  19/10/2021 autofs-5.1.8
+#  - add xdr_exports().
+diff --git a/modules/cyrus-sasl.c b/modules/cyrus-sasl.c
+index ae046e0..738e363 100644
+--- a/modules/cyrus-sasl.c
++++ b/modules/cyrus-sasl.c
+@@ -721,6 +721,10 @@ sasl_do_kinit_ext_cc(unsigned logopt, struct lookup_context *ctxt)
+ 
+ 	debug(logopt, "Kerberos authentication was successful!");
+ 
++	status = pthread_mutex_unlock(&krb5cc_mutex);
++	if (status)
++		fatal(status);
++
+ 	return 0;
+ 
+ out_cleanup_def_princ:
diff --git a/debian/patches/series b/debian/patches/series
index 932a3d3..929541a 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -10,3 +10,4 @@ spelling-error-fixes.patch
 fix-lookup-ldap-crash.patch
 fix-nfs4-mounts-in-auto-net.patch
 fix-nfs4-only-mounts-should-not-use-rpcbind.patch
+fix-missing-unlock-in-sasl_do_kinit_ext_cc.patch
